lib/ansible/modules/system/user.py

```diff
@@ -1046,12 +1046,19 @@ def ssh_key_fingerprint(self):
         return self.execute_command(cmd, obey_checkmode=False)
 
     def get_ssh_public_key(self):
-        ssh_public_key_file = '%s.pub' % self.get_ssh_key_path()
-        try:
-            with open(ssh_public_key_file, 'r') as f:
-                ssh_public_key = f.read().strip()
-        except IOError:
+        ssh_key_file = self.get_ssh_key_path()
+        if not os.path.exists(ssh_key_file):
             return None
+        cmd = [self.module.get_bin_path('ssh-keygen', True)]
+        cmd.append('-y')
+        cmd.append('-f')
+        cmd.append(ssh_key_file)
+
+        (rc, out, err) = self.execute_command(cmd, obey_checkmode=False)
```

Comment:
I don't think this can work because this is not an interactive step. So you cannot pass the key passphrase to `ssh-keygen` and the process will just stay stuck.